### DISCUSSION TITLE
训练中国象棋遇到一个问题

### DISCUSSION THREAD

**Discussion by jiangzhonglian** [apachecn/pytorch-doc-zh#573] [Q&A]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by shuangqiao999** [comment]
你好 训练中国象棋遇到一个问题 在用开局中局的fens生成的数据 显卡学习后 引擎的开局中局比较厉害 训练了残局 引擎的残局就比较厉害 开局中局又不行了 这个问题应该怎么解决？有没有办法让开局中局残局的知识融合或者通过迭代累积？

**Comment by shuangqiao999** [comment]
> 你好 训练中国象棋遇到一个问题 在用开局中局的fens生成的数据 显卡学习后 引擎的开局中局比较厉害 训练了残局 引擎的残局就比较厉害 开局中局又不行了 这个问题应该怎么解决？有没有办法让开局中局残局的知识融合或者通过迭代累积？
这是数据生成命令:
"generate_training_data depth 4 count 10000000 random_multi_pv 4 book fen.epd random_multi_pv_diff 100 random_move_count 8 random_move_max_ply 20 write_min_ply 5 write_max_ply 1000 eval_limit 10000 set_recommended_uci_options data_format bin output_file_name training_data.bin"


这是显卡学习的相关命令: "python serialize.py --features='HalfKAv2' xiangqi-6f64c55fcb28.nnue startingpointfortraining.pt"


"python train.py --resume-from-model startingpointfortraining.pt --threads 30 --num-workers 88 --gpus %gpu% --max_epochs %pochs% training_data.bin validation_data.bin "


"python serialize.py logs/default/version_0/checkpoints/last.ckpt xiangqi-1688888.nnue"
